src/Fixer/Phpdoc/PhpdocParamOrderFixer.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\Annotation;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Jonathan Gruber <gruberjonathan@gmail.com>
+ */
+final class PhpdocParamOrderFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Orders all `@param` annotations in DocBlocks according to method signature.',
+            [
+                new CodeSample(
+                    '<?php
+class C {
+    /**
+     * @param bool   $a
+     * @param string $c
+     * @param string $b
+     */
+    public function m($a, $b, $c) {}
+}
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            // Check for function signature
+            $functionSequence = $tokens->findSequence([[T_FUNCTION], [T_STRING], '('], $index);
+
+            if (null === $functionSequence) {
+                continue;
+            }
+
+            $funcParamTokens = $this->getFuncParameterTokens($tokens, $functionSequence);
+            $docBlock = $this->sortDocBlockParamAnnotations($tokens[$index], $funcParamTokens);
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $docBlock->getContent()]);
+        }
+    }
+
+    /**
+     * Fetches the tokens of a function signature.
+     *
+     * @param Tokens            $tokens
+     * @param array<int, Token> $functionSequence
+     *
+     * @return array
+     */
+    private function getFuncParameterTokens(Tokens $tokens, array $functionSequence)
```

Comment:
maybe `PhpCsFixer\Tokenizer\Analyzer\ArgumentsAnalyzer::getArguments` can be used for this?

Reply:
I checked this out. Unfortunately this function works differently. It "returns start and end token indexes of arguments". So ` $c = null` (note the leading space) leads to a single space as start and `null` as end. So in the following case you get `null` instead of `$c`:

```php
function m(array &$a, callable $b, $c = null, $d) {}

// $paramNames =
Array
(
    [8] => $a
    [13] => $b
    [18] => null
    [25] => $d
)
```

Therefore I would keep my function, since it seems to work fairly well.